pint_server/app.py

```diff
@@ -765,17 +774,110 @@ def assert_valid_date(date):
         abort(Response('', status=404))
 
 
-def make_response(content_dict, collection_name, element_name):
-    if request.path.endswith('.xml'):
-        return Response(
-            json_to_xml(content_dict, collection_name, element_name),
-            mimetype='application/xml;charset=utf-8')
+def supported_encoding(accept_encoding):
+    return (accept_encoding is not None) and any(comp_type in accept_encoding for comp_type in (
+            'bzip2', 'gzip', 'lzma'))
+
+
+def make_response_payload(
+        content_dict, collection_name, element_name, charset):
+    # generate xml or json formatted payload
+    if 'xml' in request.path:
+        payload = json_to_xml(content_dict, collection_name, element_name)
+        app_type = 'xml'
     else:
         if collection_name:
             content = {collection_name: content_dict}
         else:
             content = content_dict
-        return jsonify(**content)
+        payload = json.dumps(content, separators=(',', ':'))
+        app_type = 'json'
+
+    # encode the payload with the desired charset
+    payload = payload.encode(charset)
+    return app_type, payload
+
+
+def make_response(content_dict, collection_name, element_name):
+
+    charset = 'utf-8'
+    content_encoding_header = None
+    accept_encoding = None
```

Comment:
You can initialise accept_encoding here and avoid having to fetch it twice below

```suggestion
    accept_encoding = request.headers.get('Accept-Encoding')
```